lib/active_merchant/billing/gateways/reach.rb

```diff
@@ -152,18 +182,32 @@ def format_and_sign(post)
         post
       end
 
+      def get_network_payment_reference(response)
+        parameters = { request: { MerchantId: @options[:merchant_id], OrderId: response[:response][:OrderId] } }
+        body = post_data format_and_sign(parameters)
+
+        # unit tests doesn't work with ssl_post
+        raw_response = ssl_request :post, url('query'), body, {}
```

Comment:
💬 Comment:

For debugging and tracing this request should be part of a multiresponse object

Reply:
I could not inject the result of the 2nd request into the 1st one, so I tried this approach.